src/test/ui/associated-types/associated-types-path-2.stderr

```diff
@@ -25,13 +27,15 @@ error[E0277]: the trait bound `u32: Foo` is not satisfied
   --> $DIR/associated-types-path-2.rs:29:5
    |
 LL |     f1(2u32, 4u32);
-   |     ^^^^^^^^^^^^^^ the trait `Foo` is not implemented for `u32`
+   |     ^^ the trait `Foo` is not implemented for `u32`
```

Comment:
This is already what we had. Because of the new check, this error and the previous one got swapped. We now complain about the associated type first on the second argument, and then we complain about the materialized type itself when we evaluate the first argument. `pub fn f1<T: Foo>(a: T, x: T::A) {}` I think that because both arguments reference `T` in some way, we point at the call (as we did before this PR). I think that's reasonable, if not perfect, behavior.

Reply:
Interesting, looking at this closely, _this_ case doesn't point at the trait bound originating the obligation... It's likely that we don't have the right `ObligationCauseCode` for it.

Reply:
Resolved in the last commit

Reply:
Very nice!